robot/impl/robot_reconfigure_test.go

```diff
@@ -3754,16 +3754,64 @@ func TestResourceConstructCtxCancel(t *testing.T) {
 	})
 }
 
+func TestResourceCloseNoHang(t *testing.T) {
+	logger := logging.NewTestLogger(t)
+
+	mockAPI := resource.APINamespaceRDK.WithComponentType("mock")
+	modelName1 := utils.RandomAlphaString(5)
+	model1 := resource.DefaultModelFamily.WithModel(modelName1)
+
+	resource.RegisterComponent(mockAPI, model1, resource.Registration[resource.Resource, resource.NoNativeConfig]{
+		Constructor: func(
+			ctx context.Context,
+			deps resource.Dependencies,
+			conf resource.Config,
+			logger logging.Logger,
+		) (resource.Resource, error) {
+			return &mockFake{Named: conf.ResourceName().AsNamed(), shouldHangOnClose: true}, nil
+		},
+	})
+	defer func() {
+		resource.Deregister(mockAPI, model1)
+	}()
+
+	cfg := &config.Config{
+		Components: []resource.Config{
+			{
+				Name:  "mock",
+				Model: model1,
+				API:   mockAPI,
+			},
+		},
+	}
+	r, err := New(context.Background(), cfg, logger)
+	test.That(t, err, test.ShouldBeNil)
+
+	// Lower resourceCloseTimeout to one second to speed up test.
+	originalResourceCloseTimeout := resourceCloseTimeout
+	resourceCloseTimeout = time.Second
+	defer func() {
+		resourceCloseTimeout = originalResourceCloseTimeout
+	}()
+
+	// Assert that this Close does not hang for an hour even with
+	// time.Sleep(time.Hour) in mockFake's Close method.
+	testutils.WaitForAssertion(t, func(tb testing.TB) {
+		test.That(t, r.Close(context.Background()), test.ShouldBeNil)
+	})
```

Comment:
could we make this shorter than an hour? just to avoid having this test hang in case of a regression.

Reply:
Yep! I think that's a good idea.